R/calculateDistance.R

```diff
@@ -1,6 +1,315 @@
-# calculateDistance function is removed. This function is left for other functions 
-# to use
-.calculate_distance <- function(mat, FUN = stats::dist, ...){
-    # Distance between all samples against all samples
-    do.call(FUN, c(list(mat),list(...)))
+#' Calculate dissimilarities
+#'
+#' These functions are designed to calculate dissimilarities on data stored 
+#' within a 
+#' \code{\link[TreeSummarizedExperiment:TreeSummarizedExperiment-class]{TreeSummarizedExperiment}}
+#'  object. For overlap, Unifrac, and Jensen-Shannon Divergence (JSD) 
+#'  dissimilarities, the functions use mia internal functions, while for other 
+#'  types of dissimilarities, they rely on \code{\link[vegan:vegdist]{vegdist}}.
+#'
+#' @param x a \code{\link[TreeSummarizedExperiment:TreeSummarizedExperiment-class]{TreeSummarizedExperiment}}
+#'   object.
+#'
+#' @param method \code{Character scalar}. Specifies which distance to calculate.
+#' 
+#' @param name \code{Character scalar}. The name to be used to store the result 
+#'  in the reducedDims of the output. (Default: \code{method})
+#'
+#' @param assay.type \code{Character scalar}. Specifies which assay to use for 
+#'   calculation. (Default: \code{"counts"})
+#'
+#' @param assay_name Deprecated. Use \code{assay.type} instead.
+#'   
+#' @param exprs_values Deprecated. Use \code{assay.type} instead.
+#'   
+#' @param transposed \code{Logical scalar}. Specifies if x is transposed with cells in
+#'   rows. (Default: \code{FALSE})
+#'
+#' @param tree.name \code{Character scalar}. Specific to unifrac dissimilarity. 
+#' Specifies the name of the tree used in calculation. (Default: \code{"phylo"})
+#' 
+#' @param tree_name Deprecated. Use \code{tree.name} instead.
```

Comment:
As these are specific only for unifrac, these should be "hidden". CAn you put them to .get_tree_args()? Does it work?